lib/SIL/IR/SILProfiler.cpp

```diff
@@ -247,9 +258,10 @@ bool shouldSkipExpr(Expr *E) {
 /// Whether the children of a decl that isn't explicitly handled should be
 /// walked.
 static bool shouldWalkIntoUnhandledDecl(const Decl *D) {
-  // We want to walk into the initializer for a pattern binding decl. This
-  // allows us to map LazyInitializerExprs.
-  return isa<PatternBindingDecl>(D);
+  // We want to walk into initializers for bindings, and the expansions of
+  // MacroExpansionDecls, which will be nested within MacroExpansionExprs in
+  // local contexts.
+  return isa<PatternBindingDecl>(D) || isa<MacroExpansionDecl>(D);
```

Comment:
How do we report code coverage for the expansions of a macro? Or I don’t fully understand why we need to walk into `MacroExpansionDecl`.

Reply:
We don't, the only reason we need to walk into the expansion is to get accurate counter information for the following code, e.g for `#id(try throwingFn())`, we need to account for the fact that code after the macro may not run if an error was thrown